qiita_db/artifact.py

```diff
@@ -0,0 +1,638 @@
+
+        The parameters depend based on how the artifact was generated:
+            - If the artifact was uploaded by the user, the parameter
+            `prep_template` should be provided and the parameters `parents` and
+            `processing_parameters` should not be provided.
+            - If the artifact was generated by processing one or more
+            artifacts, the parameters `parents` and `processing_parameters`
+            should be provided and the parameter `prep_template` should not
+            be provided.
+
+        The parameters `can_be_submitted_to_ebi` and
+        `can_be_submiited_to_vamps` defaults to false and they should be
+        provided if and only if the artifact can be submitted to EBI and
+        VAMPS, respectivelly.
+
+        Parameters
+        ----------
+        filepaths : iterable of tuples (str, int)
+            A list of 2-tuples in which the first element is the artifact
+            file path and the second one is the file path type id
+        artifact_type : str
+            The type of the artifact
+        prep_template : qiita_db.metadata_template.PrepTemplate, optional
+            If the artifact is being uploaded by the user, the prep template
+            to which the artifact should be linked to. If not provided,
+            `parents` should be provided.
+        parents : iterable of qiita_db.artifact.Artifact, optional
+            The list of artifacts from which the new artifact has been
+            generated. If not provided, `prep_template` should be provided.
+        processing_parameters : qiita_db.software.Parameter, optional
+            The processing parameters used to generate the new artifact
+            from `parents`. It is required if `parents` is provided. It should
+            not be provided if `prep_template` is provided.
+        can_be_submitted_to_ebi : bool, optional
+            Whether the new artifact can be submitted to EBI or not. Default:
+            `False`.
+        can_be_submitted_to_vamps : bool, optional
+            Whether the new artifact can be submitted to VAMPS or not. Default:
+            `False`.
+
+        Returns
+        -------
+        qiita_db.artifact.Artifact
+            A new instance of Artifact
+
+        Raises
+        ------
+        QiitaDBArtifactCreationError
+            If `filepaths` is not provided
+            If both `parents` and `prep_template` are provided
+            If none of `parents` and `prep_template` are provided
+            If `parents` is provided but `processing_parameters` is not
+            If both `prep_template` and `processing_parameters` is provided
+            If not all the artifacts in `parents` belong to the same study
+
+        Notes
+        -----
+        The visibility of the artifact is set by default to `sandbox`
+        The timestamps of the artifact is set by default to `datetime.now()`
+        """
+        # We need at least one file
```

Comment:
Yeah, sometimes I first write the comments so I don't forget any case and then add the code...